volumes/miovision/sql/create-mv-mio_cent.sql

```diff
@@ -0,0 +1,158 @@
+            'River',
+            'Major Railway',
+            'Hydro Line',
+            'Walkway',
+            'Major Shoreline',
+            'Creek/Tributary',
+            'Ferry Route',
+            'Minor Railway',
+            'Minor Shoreline (Land locked)'
+        )
+
+        UNION ALL
+
+        SELECT DISTINCT
+            ter.intersection_uid,
+            ter.lat,
+            ter.lng,
+            cent.geo_id,
+            ci.latitude,
+            ci.longitude
+        FROM miovision_api.intersections AS ter
+        LEFT JOIN gis.centreline AS cent ON ter.int_id = cent.fnode
+        LEFT JOIN gis.centreline_intersection AS ci ON cent.tnode = ci.int_id
+        WHERE cent.fcode_desc NOT IN (
+            'Trail',
+            'Geostatistical line',
+            'Other',
+            'River',
+            'Major Railway',
+            'Hydro Line',
+            'Walkway',
+            'Major Shoreline',
+            'Creek/Tributary',
+            'Ferry Route',
+            'Minor Railway',
+            'Minor Shoreline (Land locked)'
+        )
+    ),
+
+    --determine difference in the "non-miovision node" co-ordinates - figure out if we're dealing with the north-south or east-west street in the intersection
+    ll_diff AS (
+        SELECT
+            ci.intersection_uid,
+            ci.lat,
+            ci.lng,
+            ci.geo_id,
+            ci.latitude,
+            ci.longitude,
+            ci.latitude - ci.lat AS lat_diff,
+            ci.longitude - ci.lng AS lng_diff,
+            CASE
+                WHEN abs(ci.latitude - ci.lat) > abs(ci.longitude - ci.lng) THEN 'NS'
+                WHEN abs(ci.longitude - ci.lng) > abs(ci.latitude - ci.lat) THEN 'EW'
+            END AS nsew
+        FROM cent_int AS ci
+    )
+
+    --determine which segments are N, S, E, W so that we can match with miovision leg + intersection_uid
+    SELECT
+        ld.geo_id AS centreline_id,
+        ld.intersection_uid,
+        CASE
+            WHEN ld.nsew = 'NS' AND ld.latitude > ld.lat THEN 'N'
+            WHEN ld.nsew = 'NS' AND ld.latitude < ld.lat THEN 'S'
+            WHEN ld.nsew = 'EW' AND ld.longitude > ld.lng THEN 'E'
+            WHEN ld.nsew = 'EW' AND ld.longitude < ld.lng THEN 'W'
+        END AS leg
+    FROM ll_diff AS ld
+);
+
+--2. It's important that the centreline to miovision file has the right combination of intersection_uids and legs. Here's how I checked that:
+WITH data_legs AS (
+    SELECT DISTINCT
+        vol.intersection_uid,
+        vol.leg
+    FROM miovision_api.volumes_15min AS vol--this gets me all the distinct intersection_uid and leg combos in the actual dataset
+)
+
+SELECT
+    cm.centreline_id,
+    cm.intersection_uid,
+    cm.leg,
+    dl.intersection_uid AS data_int,
+    dl.leg AS data_leg
+FROM data_legs AS dl
+FULL JOIN miovision_api.centerline_miovision AS cm ON cm.intersection_uid = dl.intersection_uid AND cm.leg = dl.leg -- full join means I can see what's in each pile
```

Comment:
@scann0n I'm confused that this is referencing `centerline_miovision` in the creation of `centreline_miovision`